controllers/registry_test.go

```diff
@@ -503,3 +505,116 @@ func TestSyncRegistry(t *testing.T) {
 		})
 	}
 }
+
+func TestBackOffRegistry_Get(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name   string
+		m      map[client.ObjectKey]*BackOff
+		opts   []backoff.ExponentialBackOffOpts
+		objKey client.ObjectKey
+		want   *BackOff
+		want1  bool
+	}{
+		{
+			name: "new",
+			m:    map[client.ObjectKey]*BackOff{},
+			objKey: client.ObjectKey{
+				Namespace: "foo",
+				Name:      "bar",
+			},
+			want: &BackOff{
+				bo: backoff.NewExponentialBackOff(
+					DefaultExponentialBackOffOpts()...,
+				),
+			},
+			want1: true,
+		},
+		{
+			name: "previous",
+			m: map[client.ObjectKey]*BackOff{
+				{
+					Namespace: "foo",
+					Name:      "bar",
+				}: {
+					bo: backoff.NewExponentialBackOff(
+						DefaultExponentialBackOffOpts()...,
+					),
+				},
+			},
+			objKey: client.ObjectKey{
+				Namespace: "foo",
+				Name:      "bar",
+			},
+			want: &BackOff{
+				bo: backoff.NewExponentialBackOff(
+					DefaultExponentialBackOffOpts()...,
+				),
+			},
+			want1: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &BackOffRegistry{
+				m:    tt.m,
+				opts: tt.opts,
+			}
+			got, got1 := r.Get(tt.objKey)
+			assert.NotNilf(t, got, "Get(%v)", tt.objKey)
+			assert.Equalf(t, tt.want1, got1, "Get(%v)", tt.objKey)
+			assert.Greaterf(t, got.bo.NextBackOff(), time.Duration(0), "Get(%v)", tt.objKey)
+		})
```

Comment:
Might be good to test that the backoff in the registry is being incremented by `NextBackOff()`, i.e. do another `r.Get()` and check that `NextBackoff()` is greater than the first one.

Reply:
Added in 15e14c91bacf2ebdcfc5ad3888975587fe0287c3